server/tserver/src/main/java/org/apache/accumulo/tserver/TabletClientHandler.java

```diff
@@ -1246,24 +1246,24 @@ public void loadTablet(TInfo tinfo, TCredentials credentials, String lock,
     TabletLogger.loading(extent, server.getTabletSession());
 
     final AssignmentHandler ah = new AssignmentHandler(server, extent);
-    // final Runnable ah = new LoggingRunnable(log, );
     // Root tablet assignment must take place immediately
 
     if (extent.isRootTablet()) {
-      Threads.createThread("Root Tablet Assignment", () -> {
+      // TODO KEVIN RATHBUN I think this should remain non critical. This method is ultimately
+      // called by TabletGroupWatcher.flushChanges which is always called within a loop, so will
+      // continue to retry/recreate the thread
+      Threads.createNonCriticalThread("Root Tablet Assignment", () -> {
```

Comment:
I think this should remain non critical. This method is ultimately called by TabletGroupWatcher.flushChanges which is always called within a loop, so will continue to retry/recreate the thread

Reply:
Yeah this triggered by an RPC and the manager will keep making that RPC periodically.